python/kserve/test/test_server.py

```diff
@@ -769,12 +847,25 @@ def test_unload(self, http_server_client):
 
 class TestTFHttpServerLoadAndUnLoadFailure:
     @pytest.fixture(scope="class")
-    def app(self):  # pylint: disable=no-self-use
-        server = ModelServer(
-            registered_models=DummyModelRepository(test_load_success=False)
+    def server(self):
+        server = ModelServer()
+        rest_server = RESTServer(
+            kserve_app, server.dataplane, server.model_repository_extension
+        )
+        rest_server.create_application()
+        yield server
+        kserve_app.routes.clear()
+
+    @pytest.fixture(scope="class")
+    def app(self, server):  # pylint: disable=no-self-use
+        mp = pytest.MonkeyPatch()
```

Comment:
Why do we need monkey patch ?

Reply:
Good catch. Actually I have tried a different approach before which needed this. This not needed now removing this.